### DISCUSSION TITLE
Placeholder

### DISCUSSION THREAD

**Discussion by IvanLisz** [solidjs/solid#2203] [Ideas]
I would love for `<input />` `<textarea/>` to automatically accept JSX.Element as placeholder.

**Comment by atk** [comment]
But the actual HTML attribute only accepts strings, so this change would only lead to errors. If you want placeholders with more versatile contents, you have to implement them yourself.

**Reply by IvanLisz** [reply]
I understand this can be added in userland (and maybe it should) but why? Why would it lead to errors? you could still accept a string.

**Reply by IvanLisz** [reply]
can this be done with a custom renderer?

**Reply by atk** [reply]
You can create a component that takes an input and overlays it with a span that has `pointer-events: none` set on it via css; use the span to show the JSX placeholder and hide it whenever the input has any content.
